src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -348,35 +355,38 @@ private IPropertyReferenceExpression CreateBoundPropertyAccessOperation(BoundPro
             return new LazyPropertyReferenceExpression(property, instance, member, argumentsInEvaluationOrder, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
-        private IPropertyReferenceExpression CreateBoundIndexerAccessOperation(BoundIndexerAccess boundIndexerAccess)
+        private IOperation CreateBoundIndexerAccessOperation(BoundIndexerAccess boundIndexerAccess)
         {
-            IPropertySymbol property = boundIndexerAccess.Indexer;
-            Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(boundIndexerAccess.Indexer.IsStatic ? null : boundIndexerAccess.ReceiverOpt));
-            ISymbol member = boundIndexerAccess.Indexer;
-            Lazy<ImmutableArray<IArgument>> argumentsInEvaluationOrder = new Lazy<ImmutableArray<IArgument>>(() =>
+            PropertySymbol property = boundIndexerAccess.Indexer;
+            SyntaxNode syntax = boundIndexerAccess.Syntax;
+            ITypeSymbol type = boundIndexerAccess.Type;
+            Optional<object> constantValue = ConvertToOptional(boundIndexerAccess.ConstantValue);
+            bool isImplicit = boundIndexerAccess.WasCompilerGenerated;
+
+            MethodSymbol accessor = boundIndexerAccess.UseSetterForDefaultArgumentGeneration
+                ? property.GetOwnOrInheritedSetMethod()
+                : property.GetOwnOrInheritedGetMethod();
+            
+            if (!boundIndexerAccess.OriginalIndexersOpt.IsDefault || boundIndexerAccess.ResultKind == LookupResultKind.OverloadResolutionFailure || accessor == null || accessor.OriginalDefinition is ErrorMethodSymbol)
```

Comment:
@AlekseyTs Can't figure out if `accessor.OriginalDefinition is ErrorMethodSymbol` is necessary. Leave it there just to be safe.